### PR TITLE
yolox refactored to use native blocks

### PR DESCRIPTION
**Pull request by shaydeci**
- The use of C3 replaced with YoloX CSPLayer.
- 6x6 Conv instead of Focus for Yolox.

Once we have the replacements for all the supportive objects- the next step will be to take out the if statements inside CSPDarknet constructor.

### CODE REVIEW

File: src/super_gradients/training/models/detection_models/csp_darknet53.py [hunk 1]
```diff
@@ -131,6 +131,48 @@ def forward(self, x):
         return self.cv3(torch.cat((self.m(self.cv1(x)), self.cv2(x)), dim=1))
 
 
+class CSPLayer(nn.Module):
+    """C3 in yolov5, CSP Bottleneck with 3 convolutions"""
+
+    def __init__(
```
Comment (yurkovak) [inline src/super_gradients/training/models/detection_models/csp_darknet53.py:137]: - Don't like the "n" thing.
- Activation type didn't have a default, so let's not add it, add a type hint of `Type[nn.Module]` instead.
- I think we use reStructuredText docstring everywhere in SG, but whatever you like:)

File: src/super_gradients/training/models/detection_models/csp_darknet53.py [hunk 2]
```diff
@@ -131,6 +131,48 @@ def forward(self, x):
...
+    ):
+        """
+        Args:
+            in_channels (int): input channels.
+            out_channels (int): output channels.
+            n (int): number of Bottlenecks. Default value: 1.
+        """
+        # ch_in, ch_out, number, shortcut, groups, expansion
+        super().__init__()
+        hidden_channels = int(out_channels * expansion)  # hidden channels
```
Comment (yurkovak) [inline src/super_gradients/training/models/detection_models/csp_darknet53.py:156]: Not that it matters much, but very informative comment out there 
`hidden_channels = ...  # hidden channels`
:D

File: src/super_gradients/training/models/detection_models/csp_darknet53.py [hunk 3]
```diff
@@ -232,9 +274,9 @@ def __init__(self, arch_params: HpmStruct):
         struct = [depth_mult(s) for s in struct]
         self._modules_list = nn.ModuleList()
 
-        if get_param(arch_params, 'stem_type') == 'focus' or yolo_type == 'yoloX' or yolo_version == 'v3.0':
+        if get_param(arch_params, 'stem_type') == 'focus' or yolo_version == 'v3.0':
             self._modules_list.append(Focus(channels_in, width_mult(64), 3, 1, activation_type))  # 0
-        elif get_param(arch_params, 'stem_type') == '6x6' or yolo_version == 'v6.0':
+        elif get_param(arch_params, 'stem_type') == '6x6' or yolo_type == 'yoloX' or yolo_version == 'v6.0':
```
Comment (yurkovak) [inline src/super_gradients/training/models/detection_models/csp_darknet53.py:279]: Will break yolox weights, which aren't among pretrained models tests yet, but just so you know when you add them. Let's discuss offline how to reshape them into 6x6.

**Review by yurkovak — COMMENTED** [review]
Small comment inline

File: src/super_gradients/training/models/detection_models/csp_darknet53.py (hunk 1 above)
Comment (shaydeci) [reply src/super_gradients/training/models/detection_models/csp_darknet53.py:147]: in https://github.com/Deci-AI/super-gradients/pull/248/commits/dfe766323e84dfc0085f9fe28eb50ca3a95d766e

- Renamed to num_bottlenecks.
- Removed default.
- Added docs.

**Review by yurkovak — DISMISSED** [review]
LGTM

File: src/super_gradients/training/models/detection_models/csp_darknet53.py [hunk 4]
```diff
@@ -131,6 +131,51 @@ def forward(self, x):
         return self.cv3(torch.cat((self.m(self.cv1(x)), self.cv2(x)), dim=1))
 
 
+class CSPLayer(nn.Module):
+    """
+    C3 in yolov5, CSP Bottleneck with 3 convolutions
```
Comment (ofrimasad) [inline src/super_gradients/training/models/detection_models/csp_darknet53.py:136]: please remove the comment

**Review by ofrimasad — COMMENTED** [review]
remove the comment and then LGTM

**Review by ofrimasad — APPROVED** [review]
LGTM

**Comment by shaydeci** [comment]
> LGTM

Merge is on hold- need to add support for loading the previously trained YoloX checkpoints
